src/v/datalake/table_definition.cc

```diff
@@ -15,15 +15,18 @@ namespace datalake {
 using namespace iceberg;
 struct_type schemaless_struct_type() {
     using namespace iceberg;
+    struct_type system_fields;
+    system_fields.fields.emplace_back(
+      nested_field::create(2, "offset", field_required::yes, long_type{}));
+    system_fields.fields.emplace_back(nested_field::create(
+      3, "timestamp", field_required::yes, timestamp_type{}));
+    system_fields.fields.emplace_back(
+      nested_field::create(4, "key", field_required::no, binary_type{}));
+    system_fields.fields.emplace_back(
+      nested_field::create(5, "value", field_required::no, binary_type{}));
     struct_type res;
     res.fields.emplace_back(nested_field::create(
-      1, "redpanda_offset", field_required::yes, long_type{}));
-    res.fields.emplace_back(nested_field::create(
-      2, "redpanda_timestamp", field_required::yes, timestamp_type{}));
-    res.fields.emplace_back(nested_field::create(
-      3, "redpanda_key", field_required::no, binary_type{}));
-    res.fields.emplace_back(nested_field::create(
-      4, "redpanda_value", field_required::no, binary_type{}));
+      1, "redpanda", field_required::yes, std::move(system_fields)));
```

Comment:
This is great - questions:

1. What about if the user schema has `redpanda` as value in it?
2. Should we add headers?

Reply:
Good points, at least 1 should be in this PR. I'm thinking to add 2 + partition IDs in a separate PR, though may just end up pulling them in here too.

In the proposal you suggest making this a struct called redpanda with subfield user_data, is that right? Do you mean putting it into the redpanda struct, in which case the field ordering isn't preserved?

Another approach that you might've considered, as ugly as it is, is to keep on appending _ or some other special characters until there is no more collision.

Reply:
> In the proposal you suggest making this a struct called redpanda with subfield user_data

Yes - I don't really care what the field is called there TBH.

> Do you mean putting it into the redpanda struct, in which case the field ordering isn't preserved?

Yes - we should not make any guarantees about field ordering outside of it being deterministic for a single schema. Are you saying the ordering needs to be consistent? Because I think at least the proto converter converts order based on the user specified order in the protofile. We could also put our redpanda column at the beginning if we don't replace the user column.

> Another approach that you might've considered, as ugly as it is, is to keep on appending _ or some other special characters until there is no more collision.

Ah with schema evolution this becomes really messy really fast. We either need some kind of escaping mechanism... One approach I've seen in RPCN is that `~` is a special character. If there are conflicts we append that and any user data with a `~` in it is escaped to be `~~`. I'd rather not go down this path unless we really have to.

Reply:
>Are you saying the ordering needs to be consistent?

I don't think it's a requirement, just wanted to make sure we're on the same page since it could be surprising to see, though shouldn't affect query correctness.

>Ah with schema evolution this becomes really messy really fast.

For sure, and agreed we should try to avoid if the struct nesting works.